mono/mini/interp/transform.c

```diff
@@ -828,6 +849,9 @@ interp_transform_call (TransformData *td, MonoMethod *method, MonoMethod *target
 		csignature = mono_method_signature (target_method);
 	}
 
+	if (check_visibility && target_method && !mono_method_can_access_method (method, target_method))
+		interp_generate_mae_throw (td, method, target_method);
```

Comment:
why is `check_visibility` needed?

Reply:
@lewurm we don't want to check the visibility for special calls internal to the runtime (eg Nullable:Unbox)